daemon.lua

```diff
@@ -20,7 +22,14 @@ function telegraf:push(message)
 	self:sendto(message, inet.localhost, 8094)
 end
 
-local inbox = mailbox.inbox(100 * 1024)
+local inbox = {
+	telegraf = mailbox.inbox(100 * 1024),
+	reply    = mailbox.inbox(1024)
+}
+
+function inbox:receive()
+	return self.telegraf:receive(), self.reply:receive()
+end
```

Comment:
I think it would be clearer to check individually for messages instead of encapsulating the mailboxes here..